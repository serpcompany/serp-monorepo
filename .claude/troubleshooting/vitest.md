https://vitest.dev/guide/common-errors.html