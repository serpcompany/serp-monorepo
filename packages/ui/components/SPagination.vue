<!-- eslint-disable vue/no-mutating-props -->
<script setup lang="ts">
withDefaults(
  defineProps<{
    totalItems: number
    limit: number
    siblingCount: number
  }>(),
  {
    totalItems: 0,
    limit: 50,
    siblingCount: 3,
  },
)

const page = defineModel<number>()
</script>

<template>
  <UPagination
    v-model:page="page"
    :total="totalItems"
    :items-per-page="limit"
    :sibling-count="siblingCount"
    aria-label="pagination"
    class="mt-20 flex justify-center overflow-x-auto rounded-none"
  />
</template>
