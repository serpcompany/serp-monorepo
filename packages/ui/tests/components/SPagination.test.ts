import { describe, expect, it } from 'vitest'
import SPagination from '../../components/SPagination.vue'
import ComponentRender from '../componentRender'

describe('sPagination', () => {
  it.each([
    // When no props are provided, the withDefaults in the component should kick in.
    ['with default props', { props: {} }],
    // Custom pagination settings.
    [
      'with custom pagination values',
      { props: { totalItems: 100, limit: 10, siblingCount: 2 } },
    ],
  ])('renders %s correctly', async (desc, options: { props: unknown }) => {
    const html = await ComponentRender(
      `SPagination ${desc}`,
      options,
      SPagination,
    )
    expect(html).toMatchSnapshot()
  })
})
