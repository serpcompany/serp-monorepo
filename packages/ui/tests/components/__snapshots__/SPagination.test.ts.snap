// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SPagination > renders with custom pagination values correctly 1`] = `
<nav
  aria-label="pagination"
  class="
    flex
    justify-center
    mt-20
    overflow-x-auto
    rounded-none
  "
>
  <div class="
    flex
    gap-1
    items-center
  ">
    <button
      aria-label="First Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Previous Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-current="page"
      aria-label="Page 1"
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        inline-flex
        items-center
        p-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      data-selected="true"
      data-type="page"
      type="button"
      value="1"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        1
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 2"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="2"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        2
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 3"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="3"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        3
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 4"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="4"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        4
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 5"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="5"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        5
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Next Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Last Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
  </div>
</nav>
`;

exports[`SPagination > renders with default props correctly 1`] = `
<nav
  aria-label="pagination"
  class="
    flex
    justify-center
    mt-20
    overflow-x-auto
    rounded-none
  "
>
  <div class="
    flex
    gap-1
    items-center
  ">
    <button
      aria-label="First Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Previous Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-current="page"
      aria-label="Page 1"
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        inline-flex
        items-center
        p-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      data-selected="true"
      data-type="page"
      type="button"
      value="1"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        1
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Next Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Last Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
  </div>
</nav>
`;

exports[`sPagination > renders with custom pagination values correctly 1`] = `
<nav
  aria-label="pagination"
  class="
    flex
    justify-center
    mt-20
    overflow-x-auto
    rounded-none
  "
>
  <div class="
    flex
    gap-1
    items-center
  ">
    <button
      aria-label="First Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Previous Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-current="page"
      aria-label="Page 1"
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        inline-flex
        items-center
        p-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      data-selected="true"
      data-type="page"
      type="button"
      value="1"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        1
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 2"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="2"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        2
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 3"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="3"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        3
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 4"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="4"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        4
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Page 5"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      data-type="page"
      type="button"
      value="5"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        5
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Next Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Last Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
  </div>
</nav>
`;

exports[`sPagination > renders with default props correctly 1`] = `
<nav
  aria-label="pagination"
  class="
    flex
    justify-center
    mt-20
    overflow-x-auto
    rounded-none
  "
>
  <div class="
    flex
    gap-1
    items-center
  ">
    <button
      aria-label="First Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Previous Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-left
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-current="page"
      aria-label="Page 1"
      class="
        aria-disabled:bg-primary
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-primary
        disabled:bg-primary
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:outline-2
        focus-visible:outline-offset-2
        focus-visible:outline-primary
        font-medium
        gap-1.5
        hover:bg-primary/75
        inline-flex
        items-center
        p-1.5
        rounded-md
        text-inverted
        text-sm
        transition-colors
      "
      data-selected="true"
      data-type="page"
      type="button"
      value="1"
    >
      <!-- v-if -->
      <span class="
        min-w-5
        text-center
        truncate
      ">
        1
      </span>
      <!-- v-if -->
    </button>
    <button
      aria-label="Next Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevron-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
    <button
      aria-label="Last Page"
      class="
        aria-disabled:bg-default
        aria-disabled:cursor-not-allowed
        aria-disabled:opacity-75
        bg-default
        disabled:bg-default
        disabled:cursor-not-allowed
        disabled:opacity-75
        focus-visible:ring-2
        focus-visible:ring-inverted
        focus:outline-none
        font-medium
        gap-1.5
        hover:bg-elevated
        inline-flex
        items-center
        p-1.5
        ring
        ring-accented
        ring-inset
        rounded-md
        text-default
        text-sm
        transition-colors
      "
      disabled=""
      type="button"
    >
      <span
        aria-hidden="true"
        class="
          i-lucide:chevrons-right
          iconify
          shrink-0
          size-5
        "
      ></span>
      <!-- v-if -->
      <!-- v-if -->
    </button>
  </div>
</nav>
`;
